.github/workflows/clang-format.yml

```diff
@@ -11,17 +11,14 @@ jobs:
     steps:
     - uses: actions/checkout@v2
       with:
-        # checkout PR head
-        ref: '${{github.event.pull_request.head.sha}}'
-    - name: Fetch target branch
-      run: git fetch --no-tags --prune --depth=1 origin +refs/heads/${{github.base_ref}}:refs/remotes/origin/${{github.base_ref}}
+        fetch-depth: 2
 
     - name: Get clang-format first
       run: sudo apt-get install -yqq clang-format-9
 
     - name: Run clang-format for the patch
       run: |
-        git diff -U0 --no-color origin/${{github.base_ref}}...HEAD | ./clang/tools/clang-format/clang-format-diff.py -p1 -binary clang-format-9 > ./clang-format.patch
+        git diff -U0 --no-color ${GITHUB_SHA}^1 ${GITHUB_SHA} -- | ./clang/tools/clang-format/clang-format-diff.py -p1 -binary clang-format-9 > ./clang-format.patch
```

Comment:
Am I right that this check now processes diff between the last commit in PR and one commit before? What if PR initially contains more than 1 commit?

Reply:
GITHUB_SHA - merge commit for `git merge <target branch> <PR branch>`.